x-pack/metricbeat/module/gcp/metrics/metrics_requester.go

```diff
@@ -240,6 +240,19 @@ func isAGlobalService(serviceName string) bool {
 	}
 }
 
+// isAnExistingService returns true if the given service is already an existing service supported by default in GCP module
+func isAnExistingService(serviceName string) bool {
+	switch serviceName {
+	case gcp.ServiceCompute, gcp.ServiceGKE, gcp.ServiceStorage, gcp.ServiceDataproc,
+		gcp.ServicePubsub, gcp.ServiceLoadBalancing, gcp.ServiceCloudFunctions,
+		gcp.ServiceFirestore, gcp.ServiceCloudSQL, gcp.ServiceRedis:
+		return true
+	default:
+		return false
+	}
+
+}
+
```

Comment:
I don't think we need this functionality, we can just overwrite the label if `ServiceLabel` is set in config and that should be sufficient. If we decide to set the label from the integration side due to some GCP updates or whatever, having this function would actually defeat what we’re trying to achieve in the first place.

Reply:
The reason for not allowing override in existing supported services is for the filtering to work fine by using the existing location labels. 
Yes, if GCP changes some label for these existing services at their end, we will have to do beats change for those existing services in beats module. 
We can think of not having this function, but if existing customer puts in an incorrect label, it can break for him.

Reply:
I wouldn't expose this setting to users in normal GCP integrations. I think we should control this config option and update when necessary. I think it makes sense to allow users to set it for the GCP metrics input. What do you think?

Reply:
Yeah, we can override with the label provided by the user since this is a generic input